### DISCUSSION TITLE
Requesting payment window to close after user has paid.

### DISCUSSION THREAD

**Discussion by marioaburto10** [Glench/ExtPay#133] [Ideas]
I can see that a new window or tab is opened when the function `extpay.openPaymentPage()` is called. However, when the user finishes paying, the window/tab remains open and nothing happens after that. I noticed that this loses some of my users as their journey is halted after submitting a payment. The user has to manually close the window/tab and find their way back to the window/tab that they were on to continue using the extension. This provides a layer of complexity if the user has multiple windows/tabs opened. Also, when I'm testing this functionality out, the payment windows start to stack up and I have to manually close each window one by one. I'd like to request for the payment window to close out automatically after the user pays (both in real time and in development) or at least have some text saying that they are done paying and they can head back to their previous tab and continue using the extension that they just paid for. Thank you.

**Comment by Glench** [comment]
Can you please say what you mean by "nothing happens"? For subscriptions, users are supposed to be taken to their subscription management page after payments and for one-time payments users should see a screen thanking them for paying and suggest they can safely close the payment window. Do these not happen for you?

**Reply by marioaburto10** [reply]
Thank you for the quick response. Currently, users are being taken to a management page after payments for subscriptions. However, we would like the user to be sent back to the previous tab that triggered the `extpay.openPaymentPage()` function (the active tab). Right now, my users are being taken to the manage subscription page but they would like to continue using my extension, not manage their subscription - this is where they get lost.

**Reply by Glench** [reply]
I would just add a bit in your extension content scripts that looks for the extensionpay payment page and closes it. This is not something that we'd like to add to ExtensionPay since this behavior would likely be confusing for most users.
